langchain/document_loaders/pdf.py

```diff
@@ -381,3 +381,91 @@ def load(self) -> List[Document]:
             contents = self.clean_pdf(contents)
         metadata = {"source": self.source, "file_path": self.source}
         return [Document(page_content=contents, metadata=metadata)]
+
+
+class PDFPlumberLoader(BasePDFLoader):
+    """Loader that uses PDFPlumber to load PDF files."""
+
+    def __init__(self, file_path: str):
+        """Initialize with file path."""
+        try:
+            import pdfplumber  # noqa:F401
+        except ImportError:
+            raise ValueError(
+                "PDFPlumber package not found, please install it with "
+                "`pip install pdfplumber`"
+            )
+
+        super().__init__(file_path)
+
+    def load(self, **kwargs: Optional[Any]) -> List[Document]:
+        """Load file."""
+        import pdfplumber
+
+        doc = pdfplumber.open(self.file_path)
+        file_path = self.file_path if self.web_path is None else self.web_path
+
+        return [
+            Document(
+                page_content=page.extract_text(**kwargs).encode("utf-8"),
+                metadata=dict(
+                    {
+                        "source": file_path,
+                        "file_path": file_path,
+                        "page_number": page.page_number,
+                        "total_pages": len(doc.pages),
+                    },
+                    **{
+                        k: doc.metadata[k]
+                        for k in doc.metadata
+                        if type(doc.metadata[k]) in [str, int]
+                    },
+                ),
+            )
+            for page in doc.pages
+        ]
+
+    def annotate_and_load(
+        self, folder_path: str, **kwargs: Optional[Any]
+    ) -> List[Document]:
+        """Annotate/save pdf file using pdfplumber's visual debudding and load file."""
+        path = Path(folder_path)
+        path.mkdir(exist_ok=True, parents=True)
+
+        import pdfplumber
+
+        doc = pdfplumber.open(self.file_path)
+        file_path = self.file_path if self.web_path is None else self.web_path
+
+        # get annotated PIL.Images
+        annotated_imgs = []
+        for page in doc.pages:
+            im = page.to_image(resolution=100)
+            annotated_imgs.append(im.draw_rects(page.extract_words(**kwargs)).annotated)
```

Comment:
does it make sense to be passing same kwargs in to extract_words as to extract_text? not familiar with pdfplumber so maybe it does, just wanna check

Reply:
extract_words and extract_text share some arguments, but I think you are right, we would want them separated. However, after searching a bit it seems like you cannot pass 2 separate kwargs with `def func(**kwargs1, **kwargs2)`. Is there a better convention for doing this?